### ISSUE TITLE
bids-validator checking derivatives

### ISSUE THREAD

**Issue by jokedurnez** [bids-standard/legacy-validator#366]
Hey,

Is there a reason why the bids-validator checks the derivatives?
First I had put my preprocessed data in there, which results in the bids-validator running for >2h.
Now I'm running preprocessing, and I get this error somewhere between different steps of my pipeline.
This is part of a bids-app (HCPPipelines), so I don't think I can add an ignore flag for the derivatives.

```
bids-validator /oak/stanford/groups/russpold/data/Psychosis/0.0.2//
        1: Subject label in the filename doesn't match with the path of the file. File seems to be saved in incorrect subject directory. (code: 64 - SUBJECT_LABEL_IN_FILENAME_DOESNOT_MATCH_DIRE
CTORY)

                ./derivatives/HCPPipelines/sub-S0784SQF/MNINonLinear/Native/sub-S0784SQF.ArealDistortion_FS.native.dscalar.nii

                        Evidence: File: /derivatives/HCPPipelines/sub-S0784SQF/MNINonLinear/Native/sub-S0784SQF.ArealDistortion_FS.native.dscalar.nii is saved in incorrect subject directory as

per sub-id in filename.
```

**Comment by chrisgorgo** [comment]
No - this is a bug, #346 should fix it. That branch should also have much better performance.

**Comment by chrisgorgo** [comment]
Fixed by #368
